kotlin/android/app/src/main/java/dev/firezone/android/tunnel/TunnelService.kt

```diff
@@ -307,25 +262,15 @@ class TunnelService : VpnService() {
 
     // Call this to stop the tunnel and shutdown the service, leaving the token intact.
     fun disconnect() {
-        // Acquire mutex lock
-        lock.lock()
-
-        stopNetworkMonitoring()
-
-        connlibSessionPtr?.let {
-            ConnlibSession.disconnect(it)
-        }
-
-        shutdown()
+        commandChannel?.trySend(TunnelCommand.Disconnect)
+    }
 
-        // Release mutex lock
-        lock.unlock()
```

Comment:
IIRC this mutex was added because I think we needed ordering here to prevent cases where the network monitor processed events after being disconnected. Just double-checking that moving to UniFFI removes the need to do this?

Reply:
I wasn't sure so I left it in but yes, I believe we don't need it anymore. After we have been disconnected, any future commands sent into the channel will just be dropped.